server/src/trainee/traineeController.ts

```diff
@@ -0,0 +1,187 @@
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Sets "wahl_start" and "wahl_ende" for generation
+ * @param req wahl_start and wahl_ende in form of DATETIME, generationID
+ * @param res status code and message
+ */
+export const setVotingDeadline = (req: Request, res: Response): void => {
+  database
+    .query(`UPDATE generation SET wahl_start= "?", wahl_ende= "?"  WHERE generationID=?`, [
+      req.body.wahl_start,
+      req.body.wahl_ende,
+      req.body.generationID,
+    ])
+    .then((result) => {
+      res.status(201).send("Updated Trainee Voting Deadline");
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Sets choices of internesprojekt, mentor and ressort of member
+ * @param req internesprojektID, mentorID, ressortID, mitgliedID
+ * @param res status code and message
+ */
+export const setTraineeAssignment = (req: Request, res: Response): void => {
+  database
+    .query(`UPDATE mitglied SET internesprojekt=?, mentor=?, ressort=?  WHERE mitgliedID=?`, [
+      req.body.internesprojektID,
+      req.body.mentorID,
+      req.body.ressortID,
+      req.body.mitgliedID,
+    ])
+    .then((result) => {
+      res.status(201).send("Updated Trainee Assignment");
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Addes one new member as mentor to generation
+ * @param req mitglied_ID, generation_ID
+ * @param res status code and message
+ */
+export const addMentor = (req: Request, res: Response): void => {
+  database
+    .query(`INSERT INTO generation_has_mentor (mitglied_mitgliedID, generation_generationID) VALUES (?, ?)`, [
+      req.body.mitglied_ID,
+      req.body.generation_ID,
+    ])
+    .then((result) => {
+      res.status(201).send("Added new mentor");
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Gets memberID first and last name of Mentors of generation
+ * @param req generationID
+ * @param res status code and message
+ */
+export const GetMentorsOfGeneration = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT mitgliedID, vorname, nachname, generation_generationID
+    FROM mitglied
+    INNER JOIN generation_has_mentor
+    ON generation_has_mentor.mitglied_mitgliedID = mitglied.mitgliedID
+    WHERE generation_has_mentor.generation_generationID = ?`,
+      [req.body.generationID]
+    )
+    .then((result: traineeTypes.GetMentorsOfGeneration) => {
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Gets information of internal projects of generation
+ * @param req generation_ID
+ * @param res ID, generation, name and short name
+ */
+export const GetInternalProjectsOfGeneration = (req: Request, res: Response): void => {
```

Comment:
Please use camel case for function names:
```suggestion
export const getInternalProjectsOfGeneration = (req: Request, res: Response): void => {
```

Reply:
fixed